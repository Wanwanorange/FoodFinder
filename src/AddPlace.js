import React, {Component} from 'react';

export default class AddPlace extends Component {
    constructor(props) {
        super(props);
        this.state = {
            error: undefined
        };
    }



    onInputChange = (e) => {
        e.preventDefault();
        const input = e.target.elements;
        const error = this.props.checkForErrors(input.name.value);

        this.setState(() => ({ error }));

        if (!error) {
            this.props.handleAddPlace(input.name.value, input.type.value);
            e.target.elements.name.value = '';
        }
    }

    render() {
        return (
            <div>
                <form onSubmit={this.onInputChange}>
                    <input type="text" name="name"/>
                    <select name="type">
                        <option value="bakery">Bakery</option>
                        <option value="cafe">Cafe</option>
                        <option value="meal delivery">Meal delivery</option>
                        <option value="meal takeaway">Meal Takeaway</option>
                        <option value="restaurant">Restaurant</option>
                    </select>
                    <button>Add Place</button>
                </form>
                {this.state.error && <p className="error">{this.state.error}</p>}
            </div>
        );
    }
};


