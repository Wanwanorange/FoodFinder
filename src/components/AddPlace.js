import React, {Component} from 'react';
import {addPlace} from "../actions/place-actions";
import {connect} from 'react-redux';
import selectPlaces from '../selectors/place-selector';

export class AddPlace extends Component {
    state = {
        id: '',
        name: '',
        category: 'bakery',
        completed: false,
        error: ''
    };

    addPlaceIfNoErrors = (input) => {
        const name = input.name;
        if (!name) {
            return 'Please enter place name';
        }
        else if (this.props.places.findIndex(place => place.name === name) > -1) {
            return 'This place is already on your list';
        }
        this.props.addPlace({
            id: this.state.id,
            name: this.state.name,
            category: this.state.category,
            completed: this.state.completed
        });
        return '';
    };

    onSubmit = (e) => {
        e.preventDefault();
        const error = this.addPlaceIfNoErrors(this.state);

        this.setState({error});

        if (!error) {
            document.getElementsByName('name').value = '';
        }
    };

    onNameChange = (e) => {
        const name = e.target.value;
        this.setState(() => ({ id: name, name }));
    };

    onCategoryChange = (e) => {
        const category = e.target.value;
        this.setState(() => ({ category }));
    };

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit} >
                    <input type="text" name="name" onChange={this.onNameChange}/>
                    <select name="category"
                            onChange={this.onCategoryChange}>
                        <option value="bakery">Bakery</option>
                        <option value="cafe">Cafe</option>
                        <option value="meal delivery">Meal delivery</option>
                        <option value="meal takeaway">Meal Takeaway</option>
                        <option value="restaurant">Restaurant</option>
                    </select>
                    <button>Add Place</button>
                </form>
                {this.state.error && <p className="error">{this.state.error}</p>}
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return {
        places: selectPlaces(state.places)
    };
};

const mapDispatchToProps = (dispatch) => ({
    addPlace: (place) => dispatch(addPlace(place))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddPlace);
