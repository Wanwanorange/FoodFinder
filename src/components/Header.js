import React from 'react';
import Select from 'react-select';
import { NavLink } from 'react-router-dom';
import '../styles/Header.css';

const filterOps = [
    { value: 'bakery', label: 'Bakery' },
    { value: 'cafe', label: 'Cafe' },
    { value: 'meal delivery', label: 'Meal Delivery' },
    { value: 'meal takeway', label: 'Meal Takeaway' },
    { value: 'restaurant', label: 'Restaurant' },
];

const Header = () => (
    <div>
        <header>
            <div className="tabs">
                <NavLink className="tab" activeClassName="is-active" to="/" exact={true}>Places to Go</NavLink>
                <NavLink className="tab" activeClassName="is-active" to="/completed">Places I've Been</NavLink>
            </div>
            <Select
                    className="filter"
                    isMulti
                    options={filterOps}
                    placeholder="Filter categories..."/>
        </header>
    </div>
);

export default Header;