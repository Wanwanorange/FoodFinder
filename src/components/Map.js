import React, { Component } from 'react';
import GoogleMapReact from 'google-map-react';
import { selectAll } from '../selectors/place-selector';
import { connect } from 'react-redux';
import marker from '../img/marker.png';
import completedmarker from '../img/completedmarker.png';

const AnyReactComponent = ({ completed }) => (
  <div>{completed ? <img src={completedmarker}/> : <img src={marker}/>}</div>
);

class SimpleMap extends Component {
  static defaultProps = {
    center: {
      lat: 44.5,
      lng: -85.5
    },
    zoom: 5
  };

  render() {
    return (
      <div style={{ height: '60vh', width: '100%' }}>
        <GoogleMapReact
          bootstrapURLKeys={{ key: 'AIzaSyDVpveOb7pwNDm9fll34qPkpWFdje4lxkw' }}
          defaultCenter={this.props.center}
          defaultZoom={this.props.zoom}
        >
          {this.props.places.map((place) =>
            <AnyReactComponent 
            lat={place.latitude} 
            lng={place.longitude} 
            text={place.name}
            completed={place.completed}
            />)}
        </GoogleMapReact>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    places: selectAll(state.places)
  };
};

export default connect(mapStateToProps)(SimpleMap);