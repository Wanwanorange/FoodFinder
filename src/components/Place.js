import React, { Component } from 'react';
import '../styles/Place.css';

class Place extends Component {

    removePlace = () => {
        this.props.removePlace({ id: this.props.id });
    }

    completePlace = () => {
        this.props.completePlace({ id: this.props.id });
    }

    render() {
        return (
            <div className="place">
                <div className="placeCheck">
                    <p className="placeDetails">{this.props.name}</p>
                    <p className="placeDetails">{this.props.category}</p>
                    {!this.props.completed &&
                        <button className="completeButton"
                            onClick={this.completePlace}>Been there!</button>}
                </div>
                <button className="removeButton"
                    onClick={this.removePlace} id={this.props.id}>X</button>
            </div>
        );
    }
}

export default Place