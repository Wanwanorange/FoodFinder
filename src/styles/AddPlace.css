.entryform {
  display: flex;
  justify-content: space-between;
}

.searchbar {
  font-size: 16px;
  position: relative;
  width: 70%;
}

.search-input {
  position: relative;
  width: 100%;
  height: 100%;
}

.location-search-input {
  position: absolute;
  left: 0px;
  width: 95%;
  padding: 0px 5px;
  height: 36px;
  border-radius: 5px;
  border: solid 1px lightgray;
}

.location-search-input:hover {
  border: solid 1px rgba(141, 140, 140, 0.747);
}

.location-search-input:focus {
  outline: none;
  border: solid 2px #5698fa;
}

.autocomplete-dropdown-container {
  border: solid grey 1px;
  position: absolute;
  top: 40px;
  z-index: 1;
}

.spacing {
  height: 38px;
  right: 0px;
  border-radius: 5px;
  border: solid 1px lightgray;
}

select {
  right: 100px;
  margin: 0px 5px;
  height: auto;
}

select:hover {
  border: solid 1px rgba(141, 140, 140, 0.747);
}

select:focus {
  outline: none;
  border: solid 2px #5698fa;
}

button {
  background-color: lightgray;
}

button:focus {
  outline: none;
}

button:active {
  background-color: grey;
  color: beige;
}

.Demo__clear-button {
  background: white;
  border: none;
  color: grey;
  position: absolute;
  right: 5px;
  top: 10px;
}