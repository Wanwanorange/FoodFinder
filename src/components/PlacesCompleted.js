import React, {Component} from 'react';
import Place from "./Place";

class PlacesCompleted extends Component {
    constructor(props) {
        super(props);
        this.state = {
            completedPlaces: [<Place name="Jenis Ice Cream" category="cafe"/>]
        };
    }

    render() {
        return (
            <div>
                <ul>
                    {this.state.completedPlaces}
                </ul>
            </div>
        );
    }
}

export default PlacesCompleted;