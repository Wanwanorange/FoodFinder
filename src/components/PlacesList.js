import React from 'react';
import Place from './Place';
import { removePlace, completePlace } from "../actions/place-actions";
import selectPlaces from '../selectors/place-selector';

import { connect } from 'react-redux';

export const PlacesList = (props) => (
    <div>
        {props.places.map((place) =>
            <Place {...place}
                removePlace={props.removePlace}
                completePlace={props.completePlace} />)}
    </div>
);

const mapStateToProps = (state) => {
    return {
        places: selectPlaces(state.places, false)
    };
};

const mapDispatchToProps = (dispatch) => ({
    removePlace: (data) => dispatch(removePlace(data)),
    completePlace: (data) => dispatch(completePlace(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(PlacesList);