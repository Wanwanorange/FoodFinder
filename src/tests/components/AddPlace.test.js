import React from 'react';
import { shallow } from 'enzyme';
import { AddPlace } from "../../components/AddPlace";
import places from '../testData/places';

let addPlace, wrapper, onSubmitSpy;

beforeEach(() => {
    addPlace = jest.fn();
    onSubmitSpy = jest.fn();
    wrapper = shallow(<AddPlace addPlace={addPlace}/>);
});

test('should render AddPlace correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should successfully add place with proper values', () => {
    const place = {
        id: '456',
        name: 'Sloppy Poppy',
        category: 'cafe',
        completed: false
    };
    wrapper.setState(place);
    wrapper.setProps({ places });
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });

    expect(wrapper.state('error')).toBe('');
    expect(addPlace).toHaveBeenLastCalledWith({
        id: '456',
        name: 'Sloppy Poppy',
        category: 'cafe',
        completed: false
    });
});

test('should display error if no name entry', () => {
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.state('error')).toBe('Please enter place name');
});

test('should display error if entry already exists', () => {
    const place = {
        id: '6677',
        name: 'Big Mamas',
        category: 'cafe',
        completed: false
    };
    wrapper.setState(place);
    wrapper.setProps({ places });
    wrapper.find('form').simulate('submit', {
        preventDefault: () => {}
    });
    expect(wrapper.state('error')).toBe('This place is already on your list');
});

test('should update name on input change', () => {
    const value = 'Sloppy Poppy';
    wrapper.find('input').simulate('change', {
        target: { value }
    });
    expect(wrapper.state('name')).toBe(value);
    expect(wrapper.state('id')).toBe(value);
});

test('should update category on input change', () => {
    const value = 'cafe';
    wrapper.find('select').simulate('change', {
        target: { value }
    });
    expect(wrapper.state('category')).toBe(value);
});