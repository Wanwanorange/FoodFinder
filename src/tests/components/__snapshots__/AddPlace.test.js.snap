// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddPlace correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddPlace
    addPlace={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    name="name"
                    onChange={[Function]}
                    type="text"
          />
          <select
                    name="category"
                    onChange={[Function]}
          >
                    <option
                              value="bakery"
                    >
                              Bakery
                    </option>
                    <option
                              value="cafe"
                    >
                              Cafe
                    </option>
                    <option
                              value="meal delivery"
                    >
                              Meal delivery
                    </option>
                    <option
                              value="meal takeaway"
                    >
                              Meal Takeaway
                    </option>
                    <option
                              value="restaurant"
                    >
                              Restaurant
                    </option>
          </select>
          <button>
                    Add Place
          </button>
</form>,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              type="text"
/>,
            <select
              name="category"
              onChange={[Function]}
>
              <option
                            value="bakery"
              >
                            Bakery
              </option>
              <option
                            value="cafe"
              >
                            Cafe
              </option>
              <option
                            value="meal delivery"
              >
                            Meal delivery
              </option>
              <option
                            value="meal takeaway"
              >
                            Meal Takeaway
              </option>
              <option
                            value="restaurant"
              >
                            Restaurant
              </option>
</select>,
            <button>
              Add Place
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="bakery"
>
                  Bakery
</option>,
                <option
                  value="cafe"
>
                  Cafe
</option>,
                <option
                  value="meal delivery"
>
                  Meal delivery
</option>,
                <option
                  value="meal takeaway"
>
                  Meal Takeaway
</option>,
                <option
                  value="restaurant"
>
                  Restaurant
</option>,
              ],
              "name": "category",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Bakery",
                  "value": "bakery",
                },
                "ref": null,
                "rendered": "Bakery",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cafe",
                  "value": "cafe",
                },
                "ref": null,
                "rendered": "Cafe",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Meal delivery",
                  "value": "meal delivery",
                },
                "ref": null,
                "rendered": "Meal delivery",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Meal Takeaway",
                  "value": "meal takeaway",
                },
                "ref": null,
                "rendered": "Meal Takeaway",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Restaurant",
                  "value": "restaurant",
                },
                "ref": null,
                "rendered": "Restaurant",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Place",
            },
            "ref": null,
            "rendered": "Add Place",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="name"
                        onChange={[Function]}
                        type="text"
            />
            <select
                        name="category"
                        onChange={[Function]}
            >
                        <option
                                    value="bakery"
                        >
                                    Bakery
                        </option>
                        <option
                                    value="cafe"
                        >
                                    Cafe
                        </option>
                        <option
                                    value="meal delivery"
                        >
                                    Meal delivery
                        </option>
                        <option
                                    value="meal takeaway"
                        >
                                    Meal Takeaway
                        </option>
                        <option
                                    value="restaurant"
                        >
                                    Restaurant
                        </option>
            </select>
            <button>
                        Add Place
            </button>
</form>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="name"
                onChange={[Function]}
                type="text"
/>,
              <select
                name="category"
                onChange={[Function]}
>
                <option
                                value="bakery"
                >
                                Bakery
                </option>
                <option
                                value="cafe"
                >
                                Cafe
                </option>
                <option
                                value="meal delivery"
                >
                                Meal delivery
                </option>
                <option
                                value="meal takeaway"
                >
                                Meal Takeaway
                </option>
                <option
                                value="restaurant"
                >
                                Restaurant
                </option>
</select>,
              <button>
                Add Place
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="bakery"
>
                    Bakery
</option>,
                  <option
                    value="cafe"
>
                    Cafe
</option>,
                  <option
                    value="meal delivery"
>
                    Meal delivery
</option>,
                  <option
                    value="meal takeaway"
>
                    Meal Takeaway
</option>,
                  <option
                    value="restaurant"
>
                    Restaurant
</option>,
                ],
                "name": "category",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Bakery",
                    "value": "bakery",
                  },
                  "ref": null,
                  "rendered": "Bakery",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cafe",
                    "value": "cafe",
                  },
                  "ref": null,
                  "rendered": "Cafe",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Meal delivery",
                    "value": "meal delivery",
                  },
                  "ref": null,
                  "rendered": "Meal delivery",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Meal Takeaway",
                    "value": "meal takeaway",
                  },
                  "ref": null,
                  "rendered": "Meal Takeaway",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Restaurant",
                    "value": "restaurant",
                  },
                  "ref": null,
                  "rendered": "Restaurant",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Place",
              },
              "ref": null,
              "rendered": "Add Place",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
